### PR TITLE
Added changes of Moderator, Middleware and some refactoring

### PR DESCRIPTION
**Pull request by Himan10**
* [x] Added Middleware (to verify jwt, validate request etc.) #23 
* [x] Added Moderator Actions #22 
* [x] Added update/destroy methods for company and job, and delete_user for user #19 
* [x] Added postman collections #17 (Partially done, seed files hasn't added yet)
* [x] Refactored existing code

### CODE REVIEW

File: apps/jobs/models.py [hunk 1]
```diff
@@ -112,8 +116,9 @@ class Meta:
     age = models.PositiveIntegerField(default=None, null=True)
     education = models.TextField(max_length=500, default=None, null=True)
     professional_skills = models.TextField(max_length=500, default=None, null=True)
-    hiring_status = models.CharField(max_length=15, choices=HIRING_STATUS, default="Not Applied Yet", null=True)
-
+    hiring_status = models.CharField(
+        max_length=15, choices=HIRING_STATUS, default="Not Applied Yet", null=True
```
Comment (HimanshuSharma-Angel) [inline apps/jobs/models.py:120]: instead of not applied lets store null

**Review by HimanshuSharma-Angel — COMMENTED** [review]
will review later

File: apps/accounts/views.py [hunk 2]
```diff
@@ -464,3 +465,27 @@ class RestrictedPage(APIView):
...
+class Moderator(BasePermission):
+    """
+    This class contains everything related to operations
+    belong to Moderator. Moderator user is different from
+    Admin user, but has some level of responsibilities and
+    access to resources.
+    """
+
+    def has_permission(self, request, *args):
+        """Method to check if the given user_id belongs to moderator or not"""
```
Comment (hims1911) [inline apps/accounts/views.py:479]: this code can be optimised to this,

```
    def has_permission(self, request, *args):
        """Method to check if the given user_id belongs to a moderator or not"""

        user_model = get_user_model()

        # check if user_id is valid or contains an improper value
        if not request.user_id or not is_valid_uuid(request.user_id):
            return False

        # check if the given user_id is present or not, if present then check if it's a moderator
        try:
            user = user_model.objects.get(id=request.user_id)
            return user.is_moderator
        except user_model.DoesNotExist:
            return False
```

Imported get_user_model() to dynamically fetch the user model. This is a more flexible approach as it adapts to any custom user model.

File: apps/jobs/serializers.py [hunk 3]
```diff
@@ -73,6 +73,40 @@ class Meta:
         model = Company
         fields = "__all__"
 
+    def to_representation(self, instance):
```
Comment (hims1911) [inline apps/jobs/serializers.py:76]: optimisation required

**Comment by Himan10** [comment]
I've removed the moderator actions from the urls.py, will do the other requested changes.

**Review by hims1911 — APPROVED** [review]

**Pull request merged**